### ISSUE TITLE
Version 0.32.0 broke Rubocop formatting when .rubocop.yml file does not already exist

### ISSUE THREAD

**Issue by escobera** [Glavin001/atom-beautify#2066]
# Description

After the upgrade to 0.32.0 I can't format ruby files anymore. It shows the running message, but does nothing to the code.

If I downgrade to 0.30.9 it works again.

# Steps to Reproduce

1. Add code to Atom editor
2. Run command `Atom Beautify: Beautify Editor`
3. This beautified code does not look right!

# Debug

Here is a link to the `debug.md` Gist: https://gist.github.com/escobera/84f2f0218297c5dd8c9f5b5b0d9c7e2a

# Checklist

I have:
- [x] Tried uninstalling and reinstalling Atom Beautify to ensure it installed properly
- [x] Reloaded (or restarted) Atom to ensure it is not a caching issue
- [x] Searched through existing Atom Beautify Issues at https://github.com/Glavin001/atom-beautify/issues
so I know this is not a duplicate issue
- [x] Generated debugging information by executing `Atom Beautify: Help Debug Editor` command in Atom and added link for `debug.md` Gist to this issue

**Comment by Glavin001** [comment]
cc @bbugh for https://github.com/Glavin001/atom-beautify/pull/1776 

Looking at @escobera 's logs:

```
2018-03-04T16:47:33.798Z - debug: [] spawn /home/rafa/.rvm/gems/ruby-2.1.5/bin/rubocop 0=--auto-correct, 1=--force-exclusion, 2=--stdin, 3=atom-beautify.rb, 4=--config, 5=rubocop-config11824-19981-imx77h.6go2hxs9k9
2018-03-04T16:47:34.619Z - debug: [] spawn done 2 Configuration file not found: /home/rafa/projects/athena/rubocop-config11824-19981-imx77h.6go2hxs9k9
 
2018-03-04T16:47:34.619Z - verbose: [] spawn result, returnCode 2
```

Does the message `Configuration file not found: /home/rafa/projects/athena/rubocop-config11824-19981-imx77h.6go2hxs9k9` mean anything to Rubocop users (I am not one)?

cc @szeck87

**Comment by szeck87** [comment]
The console logs give a better clue as to what's going on.  It's looking for the config file in the path to where you are working versus the temp file location.  See screens below to compare before and after the change.  This is happening due to this line being passed into the exe:  https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/rubocop.coffee#L56

![old](https://user-images.githubusercontent.com/8315038/36948672-040dac08-1fa4-11e8-93bf-a60888958ac1.png)
![new](https://user-images.githubusercontent.com/8315038/36948671-03fedade-1fa4-11e8-8199-1363ef871d77.png)

**Comment by bbugh** [comment]
Sorry about that @escobera, I tested it with several different configurations but it looks like I missed one.

@szeck87 Yes, that `projectPath` is required because of Rubocop's incredibly convoluted configuration matching system. 

It looks to me like @escobera doesn't have a `.rubocop.yml` anywhere in the parent tree starting at the `projectPath`, because it's creating a temporary Rubocop config file. This was legacy behavior from the previous setup of atom-beautify's Rubocop, which creates a temporary `.rubocop.yml` file if they don't have one in their project. Unfortunately, this is a hack and can lead to all sorts of odd behavior (because of Rubocop's incredibly convoluted configuration system). 

A better way for `atom-beautify` to handle the case of a user not having a `.rubocop.yml` file would be to simply use the default Rubocop behavior, rather than creating a temp file based on the indentation option in the beautify config. If the user wants it to behave differently, they would make a `.rubocop.yml` file. This is more in line with how Rubocop is intended to used. Anything we do to support the user not having a `.rubocop.yml` anywhere in their tree is going to be kludgy, no matter what.

@escobera can you confirm that you don't have a `.rubocop.yml` in your project path, in your home folder, or somewhere further up the tree?

**Comment by escobera** [comment]
@bbugh yeah, I can confirm that. No rubocop.yml file anywhere. I just go with the defaults.

**Comment by szeck87** [comment]
In that case, the simple fix would just be to remove the line that adds the default config:  https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/rubocop.coffee#L51.  You'll lose the indentation though.

**Comment by Glavin001** [comment]
> A better way for atom-beautify to handle the case of a user not having a .rubocop.yml file would be to simply use the default Rubocop behavior, rather than creating a temp file based on the indentation option in the beautify config. If the user wants it to behave differently, they would make a .rubocop.yml file. This is more in line with how Rubocop is intended to used. Anything we do to support the user not having a .rubocop.yml anywhere in their tree is going to be kludgy, no matter what.

Abstracting away how Rubocop and other beautifiers load configuration is an important objective of Atom-Beautify. Allowing users to use Atom-Beautify's own configuration options to configure a beautifier without a beautifier's own configuration file is a critical feature. It is unfortunate Rubocop does not have a nice configuration system, however, Atom-Beautify should still try to find a workaround the best it can. This needs to be fixed such that @escobera will work without any changes, such as creating a new `.rubocop.yml` file.

**Comment by bbugh** [comment]
Great info, I'm happy to have a deeper knowing of atom-beautify's intent. Thanks @Glavin001!

**Comment by Glavin001** [comment]
Published to v0.32.2
